sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionProcessor.java

```diff
@@ -3,29 +3,67 @@
 
 package com.azure.messaging.eventhubs;
 
+import com.azure.core.util.logging.ClientLogger;
 import com.azure.messaging.eventhubs.models.PartitionContext;
+import java.util.Objects;
 import reactor.core.publisher.Mono;
 
 /**
- * The interface defining all the operations that must be supported by a single partition processor.
+ * An abstract class defining all the operations that a partition processor can perform.
  * <p>
  * An instance of partition processor will process events only from a single partition. New instances of partition
  * processors will be created through {@link PartitionProcessorFactory#createPartitionProcessor(PartitionContext,
  * CheckpointManager) PartitionProcessorFactory}.
  * </p>
  * <p>
- * Implementations of this interface also have the responsibility of updating checkpoints when appropriate.
+ * Implementations of this abstract class also have the responsibility of updating checkpoints when appropriate.
  * </p>
  */
-public interface PartitionProcessor {
+public abstract class PartitionProcessor {
+
+    private final ClientLogger logger = new ClientLogger(PartitionProcessor.class);
+    private final PartitionContext partitionContext;
+    private final CheckpointManager checkpointManager;
+
+    /**
+     * Creates a new instance of PartitionProcessor with the given partition context and checkpoint manager
+     *
+     * @param partitionContext The partition information specific to this PartitionProcessor instance.
+     * @param checkpointManager The checkpoint manager this PartitionProcessor can use to update checkpoints.
+     */
+    public PartitionProcessor(PartitionContext partitionContext, CheckpointManager checkpointManager) {
```

Comment:
As discussed offline, please consider
- passing partition context in the individual methods instead of the constructor
- having the updateCheckpoint() methods in the partition context and not exposing CheckpointManager to the user